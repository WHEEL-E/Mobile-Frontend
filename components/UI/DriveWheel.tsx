import React from "react";
import { Dimensions, StyleSheet, TouchableOpacity, View } from "react-native";
import { Ionicons } from "@expo/vector-icons";
import { LinearGradient } from "expo-linear-gradient";
import colors from "../../constants/colors";

const SCREEN_WIDTH = Dimensions.get("screen").width;

const DriveWheel = () => {
  return (
    <View style={styles.background}>
      <LinearGradient
        colors={["#000000", "#494949", "#000000"]}
        style={styles.linearGradient}
      >
        <TouchableOpacity style={styles.arrow}>
          <Ionicons name="chevron-up-outline" color="white" size={30} />
        </TouchableOpacity>
        <View style={styles.sideArrowsContainer}>
          <TouchableOpacity style={styles.arrow}>
            <Ionicons name="chevron-back-outline" color="white" size={30} />
          </TouchableOpacity>
          <LinearGradient
            colors={["#000000", "#000000", "#949494"]}
            style={styles.secondLinearGradient}
          >
            <LinearGradient
              colors={["#949494", "#000000", "#000000"]}
              style={styles.ThirdLinearGradient}
            >
              <LinearGradient
                colors={["#949494", "#000000", "#000000", "#949494"]}
                style={styles.fourthLinearGradient}
              >
                <View style={styles.whiteDot} />
              </LinearGradient>
            </LinearGradient>
          </LinearGradient>
          <TouchableOpacity style={styles.arrow}>
            <Ionicons name="chevron-forward-outline" color="white" size={30} />
          </TouchableOpacity>
        </View>
        <TouchableOpacity style={styles.arrow}>
          <Ionicons name="chevron-down-outline" color="white" size={30} />
        </TouchableOpacity>
      </LinearGradient>
    </View>
  );
};

const styles = StyleSheet.create({
  arrow: {
    alignSelf: "center",
  },
  sideArrowsContainer: {
    flexDirection: "row",
  },
  background: {
    borderRadius: SCREEN_WIDTH * 0.45,
    width: SCREEN_WIDTH * 0.9,
    height: SCREEN_WIDTH * 0.9,
    backgroundColor: colors.lightPurple,
    justifyContent: "center",
    alignItems: "center",
    marginTop: 50,
    marginBottom: 30,
  },
  linearGradient: {
    borderRadius: SCREEN_WIDTH * 0.4,
    width: SCREEN_WIDTH * 0.8,
    height: SCREEN_WIDTH * 0.8,
    justifyContent: "center",
    alignItems: "center",
  },
  secondLinearGradient: {
    borderRadius: SCREEN_WIDTH * 0.3,
    width: SCREEN_WIDTH * 0.6,
    height: SCREEN_WIDTH * 0.6,
    justifyContent: "center",
    alignItems: "center",
  },
  ThirdLinearGradient: {
    borderRadius: SCREEN_WIDTH * 0.2,
    width: SCREEN_WIDTH * 0.4,
    height: SCREEN_WIDTH * 0.4,
    justifyContent: "center",
    alignItems: "center",
  },
  fourthLinearGradient: {
    borderRadius: SCREEN_WIDTH * 0.1,
    width: SCREEN_WIDTH * 0.2,
    height: SCREEN_WIDTH * 0.2,
    justifyContent: "center",
    alignItems: "center",
  },
  whiteDot: {
    borderRadius: SCREEN_WIDTH * 0.0125,
    width: SCREEN_WIDTH * 0.025,
    height: SCREEN_WIDTH * 0.025,
    backgroundColor: "white",
  },
});

export default DriveWheel;
