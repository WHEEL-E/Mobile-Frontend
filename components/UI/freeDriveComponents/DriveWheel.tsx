import React from "react";
import { StyleSheet, View } from "react-native";
import { Svg } from "react-native-svg";
import { DEVICE_WIDTH } from "../../../constants/dimentions";
import colors from "../../../constants/colors";
import { DriveWheelButtons } from "./DriveWheelButton";
import { DriveWheelInnershape } from "./DriveWheelInnershape";

const DriveWheel = () => {
  const buttons = ["right", "up", "left", "down"];

  return (
    <View style={styles.container}>
      <Svg
        width="100%"
        height="100%"
        style={{ shadowColor: "white", shadowRadius: 4 }}
      >
        {buttons.map((value: string, index: number) => (
          <DriveWheelButtons index={index} value={value} />
        ))}
        <DriveWheelInnershape />
      </Svg>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    borderRadius: DEVICE_WIDTH * 0.45,
    width: DEVICE_WIDTH * 0.9,
    height: DEVICE_WIDTH * 0.9,
    backgroundColor: colors.offWhite,
    marginTop: 50,
    marginBottom: 30,
  },
});

export default DriveWheel;
