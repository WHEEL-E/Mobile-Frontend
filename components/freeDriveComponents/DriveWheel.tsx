import React from "react";
import { StyleSheet, View } from "react-native";
import { Svg } from "react-native-svg";
import { DEVICE_WIDTH } from "../../utilities/constants/dimentions";
import colors from "../../utilities/constants/colors";
import { DriveWheelButton } from "./DriveWheelButton";
import { DriveWheelInnershape } from "./DriveWheelInnershape";

const DriveWheel = () => {
  const buttons = ["right", "up", "left", "down"];

  return (
    <View style={styles.container}>
      <Svg width="100%" height="100%">
        {buttons.map((value: string, index: number) => (
          <DriveWheelButton index={index} value={value} />
        ))}
        <DriveWheelInnershape />
      </Svg>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    borderRadius: DEVICE_WIDTH * 0.45,
    width: DEVICE_WIDTH * 0.9,
    height: DEVICE_WIDTH * 0.9,
    backgroundColor: colors.offWhite,
    marginTop: 50,
    marginBottom: 30,
  },
});

export default DriveWheel;
