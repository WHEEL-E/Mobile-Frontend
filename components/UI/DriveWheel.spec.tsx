import React from "react";
import renderer from "react-test-renderer";
import { render } from "@testing-library/react-native";
import { Ionicons } from "@expo/vector-icons";
import DriveWheel from "./DriveWheel";

jest.mock("@expo/vector-icons");

describe("driveWheel", () => {
  it("make sure the component matches the snapshot", () => {
    const driveWheel = renderer.create(<DriveWheel />).toJSON();
    expect(driveWheel).toMatchSnapshot();
  });

  it("makes sure the arrows are called", () => {
    render(<DriveWheel />);
    expect(Ionicons).toBeCalledTimes(4);
  });
});
