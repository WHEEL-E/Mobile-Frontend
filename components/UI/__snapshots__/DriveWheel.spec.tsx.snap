// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`driveWheel make sure the component matches the snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#E7D9EA",
      "borderRadius": 337.5,
      "height": 675,
      "justifyContent": "center",
      "marginBottom": 30,
      "marginTop": 50,
      "width": 675,
    }
  }
>
  <ViewManagerAdapter_ExpoLinearGradient
    proxiedProperties={
      Object {
        "colors": Array [
          4278190080,
          4282992969,
          4278190080,
        ],
        "endPoint": undefined,
        "locations": undefined,
        "startPoint": undefined,
      }
    }
    style={
      Object {
        "alignItems": "center",
        "borderRadius": 300,
        "height": 600,
        "justifyContent": "center",
        "width": 600,
      }
    }
  >
    <View
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignSelf": "center",
          "opacity": 1,
        }
      }
    />
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignSelf": "center",
            "opacity": 1,
          }
        }
      />
      <ViewManagerAdapter_ExpoLinearGradient
        proxiedProperties={
          Object {
            "colors": Array [
              4278190080,
              4278190080,
              4287927444,
            ],
            "endPoint": undefined,
            "locations": undefined,
            "startPoint": undefined,
          }
        }
        style={
          Object {
            "alignItems": "center",
            "borderRadius": 225,
            "height": 450,
            "justifyContent": "center",
            "width": 450,
          }
        }
      >
        <ViewManagerAdapter_ExpoLinearGradient
          proxiedProperties={
            Object {
              "colors": Array [
                4287927444,
                4278190080,
                4278190080,
              ],
              "endPoint": undefined,
              "locations": undefined,
              "startPoint": undefined,
            }
          }
          style={
            Object {
              "alignItems": "center",
              "borderRadius": 150,
              "height": 300,
              "justifyContent": "center",
              "width": 300,
            }
          }
        >
          <ViewManagerAdapter_ExpoLinearGradient
            proxiedProperties={
              Object {
                "colors": Array [
                  4287927444,
                  4278190080,
                  4278190080,
                  4287927444,
                ],
                "endPoint": undefined,
                "locations": undefined,
                "startPoint": undefined,
              }
            }
            style={
              Object {
                "alignItems": "center",
                "borderRadius": 75,
                "height": 150,
                "justifyContent": "center",
                "width": 150,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "white",
                  "borderRadius": 9.375,
                  "height": 18.75,
                  "width": 18.75,
                }
              }
            />
          </ViewManagerAdapter_ExpoLinearGradient>
        </ViewManagerAdapter_ExpoLinearGradient>
      </ViewManagerAdapter_ExpoLinearGradient>
      <View
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignSelf": "center",
            "opacity": 1,
          }
        }
      />
    </View>
    <View
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignSelf": "center",
          "opacity": 1,
        }
      }
    />
  </ViewManagerAdapter_ExpoLinearGradient>
</View>
`;
